### PR TITLE
🐛 fix: check for nil matching subnet when publicIP: true

### PR DESCRIPTION
**Pull request by damdo**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
When `publicIP: true` the controller can panic due to nil pointer dereference in case the filtered subnets are not among the ones defined under `AWSCluster.Spec.NetworkSpec.Subnets`.

**Release note**:

```release-note
fix: check for nil matching subnet when publicIP: true
```

/cc. @r4f4

### CODE REVIEW

**Review by r4f4 — COMMENTED** [review]
/lgtm

Thanks for fixing this and I'm sorry for the oversight.

File: pkg/cloud/services/ec2/instances.go [hunk 1]
```diff
@@ -352,9 +352,17 @@ func (s *Service) findSubnet(scope *scope.MachineScope) (string, error) {
 					*subnet.SubnetId, *subnet.AvailabilityZone, *failureDomain)
 				continue
 			}
-			if scope.AWSMachine.Spec.PublicIP != nil && *scope.AWSMachine.Spec.PublicIP && !s.scope.Subnets().FindByID(*subnet.SubnetId).IsPublic {
-				errMessage += fmt.Sprintf(" subnet %q is a private subnet.", *subnet.SubnetId)
-				continue
+
+			if scope.AWSMachine.Spec.PublicIP != nil && *scope.AWSMachine.Spec.PublicIP {
```
[the remark this answers is not in the document]
Comment (damdo) [reply pkg/cloud/services/ec2/instances.go]: Good suggestion. Added.

**Review by alexander-demicev — APPROVED** [review]
/lgtm

**Pull request merged**